client/extensions/wp-super-cache/cache-stats.jsx

```diff
@@ -56,23 +86,49 @@ class CacheStats extends Component {
 						</tr>
 					</thead>
 					<tbody>
-						{ map( files, ( file, dir ) => (
-							<tr className="wp-super-cache__stat" key={ dir }>
-								<td className="wp-super-cache__stat-dir">{ dir }</td>
-								<td>{ file.files }</td>
-								<td className="wp-super-cache__stat-age">{ getAge( file ) }</td>
-								<td className="wp-super-cache__stat-action">
-									<Button compact>
-										{ translate( 'Delete' ) }
-									</Button>
-								</td>
-							</tr>
-						) ) }
+						{ map( files, ( { files: count, lower_age, upper_age }, url ) => (
+						<tr className="wp-super-cache__stat" key={ url }>
+							<td className="wp-super-cache__stat-dir">
+								{ url }
+							</td>
+							<td>
+								{ count }
+							</td>
+							<td className="wp-super-cache__stat-age">
+								{ getAge( lower_age, upper_age ) }
+							</td>
+							<td className="wp-super-cache__stat-action">
+								<Button
+									compact
+									busy={ isDeleting && ( this.state.url === url ) }
+									data-url={ url }
```

Comment:
Actually, the React pattern I'm more familiar with is to bind `this.deleteFile` to `url` inside this loop, thus bypassing the need for the `data-url` attribute and reading from `event.currentTarget.dataset.url`.